behaviour/graql/reasoner/relation-inference.feature

```diff
@@ -0,0 +1,1299 @@
+#
+# Copyright (C) 2020 Grakn Labs
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+Feature: Relation Inference Resolution
+
+  Background: Set up keyspaces for resolution testing
+
+    Given connection has been opened
+    Given connection delete all keyspaces
+    Given connection open sessions for keyspaces:
+      | materialised |
+      | reasoned     |
+    Given materialised keyspace is named: materialised
+    Given reasoned keyspace is named: reasoned
+    Given for each session, graql define
+      """
+      define
+
+      person sub entity,
+        has name,
+        plays friend,
+        plays employee;
+
+      company sub entity,
+        has name,
+        plays employer;
+
+      place sub entity,
+        has name,
+        plays location-subordinate,
+        plays location-superior;
+
+      friendship sub relation,
+        relates friend;
+
+      employment sub relation,
+        relates employee,
+        relates employer;
+
+      location-hierarchy sub relation,
+        relates location-subordinate,
+        relates location-superior;
+
+      name sub attribute, value string;
+      """
+
+
+  ##############################
+  # CONDITIONS AND CONCLUSIONS #
```

Comment:
hmmmmmm

Reply:
Test basic relation inference

Reply:
or something else that is better than both

Reply:
OK basic functionality it is